### PR TITLE
RFC003 --> Han/Herc20

### PR DESCRIPTION
**Pull request by D4nte**
Work towards #1953

### CODE REVIEW

**Comment by D4nte** [comment]
@thomaseizinger I created an end-of-day draft PR to see if you have any inputs.

I went for a bottom up approach and see how it would go.

My next step here would be to split the `LedgerState` in 2, one for `han` and one for `herc20`.

File: cnd/src/swap_protocols/rfc003/create_swap.rs [hunk 1]
```diff
@@ -293,15 +288,23 @@ where
...
-    AlphaDeployed(Deployed<AL>),
-    AlphaFunded(Funded<AL, AA>),
-    AlphaRedeemed(Redeemed<AL>),
-    AlphaRefunded(Refunded<AL>),
-
-    BetaDeployed(Deployed<BL>),
-    BetaFunded(Funded<BL, BA>),
-    BetaRedeemed(Redeemed<BL>),
-    BetaRefunded(Refunded<BL>),
+    AlphaHanFunded(han::Funded<AL, AA>),
```
Comment (thomaseizinger) [inline cnd/src/swap_protocols/rfc003/create_swap.rs]: That is a bit of a bummer but I think still the right trade off in moving forward :)

**Comment by D4nte** [comment]
@tcharding base branch changed to dev.

**Comment by thomaseizinger** [comment]
I am gonna close this for now. Thanks for the effort @D4nte and @tcharding !

**Pull request closed without merging**